src/sidebar/components/ShareDialog/test/FileInput-test.js

```diff
@@ -0,0 +1,101 @@
+import { mount } from 'enzyme';
+import { createRef } from 'preact';
+
+import FileInput from '../FileInput';
+
+describe('FileInput', () => {
+  let fakeOnFileSelected;
+  let fileInputRef;
+
+  beforeEach(() => {
+    fakeOnFileSelected = sinon.stub();
+    fileInputRef = createRef();
+  });
+
+  afterEach(() => {
+    fileInputRef.current.click.restore();
+  });
+
+  const createInput = (disabled = undefined) => {
+    const wrapper = mount(
+      <FileInput
+        onFileSelected={fakeOnFileSelected}
+        fileInputRef={fileInputRef}
+        disabled={disabled}
+      />,
+    );
+
+    // Mock ref's current after mounting element, otherwise fileInputRef.current
+    // is still null
+    sinon.stub(fileInputRef.current, 'click');
```

Comment:
In my testing, `fileInputRef.current` _is_ set immediately after `mount` is called. It looks to me like what the code is actually doing is stubbing the file input's `click` method so it doesn't show a real file dialog. That makes sense, but the comments need to reflect that. The stubbing could also be done by either:

1. Finding the file input element file via a selector, rather than using an exposed ref. This is the simplest approach.
2. Using the `elementRef` pattern we use in the frontend shared package to expose the internal ref, but only use it in the tests. The @hypothesis/frontend-shared package has a `useSyncedRef` hook for this purpose.

Either way, the comment needs to be updated to reflect reality.

Reply:
Oh! I ended up in `useSyncedRef` code a couple of times without fully understanding what it does. Now I get it and it's pretty useful.

However, in this case you are probably right that finding the input via DOM selector just in the test, makes more sense, because there's no use on exposing the ref as part of the component props. It just makes it more confusing.